### PR TITLE
Update README to provide a better explanation of setting up

### PR DESCRIPTION
**Pull request by danielbachhuber**
From https://github.com/wp-cli/handbook/pull/470

### CODE REVIEW

File: .readme-partials/INSTALLATION.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-```
-
-This will:
+In order to make code changes to WP-CLI, you'll need to set up this `wp-cli-dev` development environment on your local machine. The setup process will:
 
 1. Clone all relevant packages from the `wp-cli` GitHub organization into the `wp-cli-dev` folder, and
 2. Install all Composer dependencies for a complete `wp-cli-bundle` setup, while symlinking all of the previously cloned packages into the Composer `vendor` folder.
 3. Symlink all folder in `vendor` into corresponding `vendor` folders in each repository, thus making the centralized functionality based on Composer available in each repository subfolder.
+
+First, clone the GitHub repository:
```
Comment (wojtekn) [inline .readme-partials/INSTALLATION.md:9]: I like the details you've added. I'm wondering if we could restructure it to keep only one list of steps. When I read the current version, after reading the first bullet I open the terminal and start thinking about how to clone all packages from the organization. It's not fully clear if I should perform steps 1-3 and then follow the steps below, or if the steps below are detailed instructions for steps 1-3. If the latter is true, could we compile them in one list that explains what should be done in a given step and include a sample command?

File: .readme-partials/INSTALLATION.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+
+```bash
+git clone https://github.com/wp-cli/wp-cli-dev wp-cli-dev
+```
+
+Before you can proceed further, you'll need to make sure you have [Composer](https://getcomposer.org/), PHP, and a functioning MySQL or MariaDB server on your local machine too.
```
Comment (swissspidy) [inline .readme-partials/INSTALLATION.md:15]: Do we want to mention experimental SQLite support here?

Comment (danielbachhuber) [reply .readme-partials/INSTALLATION.md:15]: @swissspidy I'm open to it! What language would you suggest?

File: .readme-partials/INSTALLATION.md (hunk 1 above)
Comment (danielbachhuber) [reply .readme-partials/INSTALLATION.md]: Fair enough, I consolidated to a single code block with 056411e

File: .readme-partials/INSTALLATION.md (hunk 2 above)
Comment (swissspidy) [reply .readme-partials/INSTALLATION.md]: Maybe something along these lines:

> Alternatively, if you do not have MySQL or MariaDB installed on your computer, you can also run tests using SQLite, which is an alternative database that does not require separate installation. To do so, set the `WP_CLI_TEST_DBTYPE` environment variable accordingly when running tests. Example: `WP_CLI_TEST_DBTYPE=sqlite composer behat`
Note: In this case you do not need to run `composer prepare-tests` either.


Needs some pointer to the Behat docs I suppose.

Comment (danielbachhuber) [reply .readme-partials/INSTALLATION.md]: I'll add these in a moment.

**Pull request merged**